java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/Snapshot.java

```diff
@@ -61,8 +77,51 @@ public Snapshot scenario(String scenario) {
      * @return Snapshot
      */
     public Snapshot serializer(SnapshotSerializer serializer) {
-        this.snapshotSerializer = serializer;
-        return this;
+        return this.withSnapshotSerializer(serializer);
+    }
+
+    /**
+     * Apply a custom comparator for this snapshot
+     *
+     * @param comparator your custom comparator
+     * @return Snapshot
+     */
+    public Snapshot comparator(SnapshotComparator<?> comparator) {
+        return this.withSnapshotComparator(comparator);
+    }
+
+    /**
+     * Apply a list of custom diff reporters for this snapshot
+     * This will replace the default reporters defined in the config
+     *
+     * @param diffReporters your custom diff reporters
+     * @return Snapshot
+     */
+    public Snapshot reporters(List<SnapshotDiffReporter> diffReporters) {
+        return this.withSnapshotDiffReporters(diffReporters);
+    }
+
+    /**
+     * Apply a single custom diff reporter for this snapshot
+     * This will replace the default reporters defined in the config
+     *
+     * @param diffReporter your custom diff reporter
+     * @return Snapshot
+     */
+    public Snapshot reporter(SnapshotDiffReporter diffReporter) {
+        return this.withSnapshotDiffReporters(Collections.singletonList(diffReporter));
+    }
+
+    /**
+     * Add a single custom diff reporter to this snapshot
+     *
+     * @param diffReporter your custom diff reporter
+     * @return Snapshot
+     */
+    public Snapshot extraReporter(SnapshotDiffReporter diffReporter) {
```

Comment:
I think this method in unnecessary. 

This can be done in `reporter ` if we use `.add()`.  That way we have a single method

```
expect("blah")
.reporter(Foo.class)
.reporter(Bar.class)
.toMatchSnapshot();
```

If we want to override the lot perhaps we can introduce `reporters` to as a varargs:
```
expect("blah")
.reporters(Foo.class, Bar.class)
.toMatchSnapshot();
``

Reply:
Hmm. I wanted to have the 'reporter' method replace all existing reporters. That means if we chain the calls, only the last one will take effect. I'm not sure if changing 'reporter' method to be additive is the right approach since the most common use of this method would probably be to replace the default reporter with a custom one. If we make the method additive, users would have to explicitly clear existing default reporters and then add their custom one.  

I think a varargs reporters makes sense. So I'll remove the 'extraReporters' method and change the existing 'reporters' method to accept a varargs reporters instead of a list. 

Does that sound good ?

Reply:
Yep, sounds good to me